lib/ruby_indexer/lib/ruby_indexer/prefix_tree.rb

```diff
@@ -0,0 +1,103 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class PrefixTree
```

Comment:
I wonder if we should make this generic and have nodes point to index entries instead?

That way, with features such as autocompletion `search` would return an array of index entries and you could get the list of names matching plus show comments, signatures or other metadata?

Reply:
Since this data structure is going to be a cornerstone of the indexer, I would also add a comment explaining what it does, how to use it and a few examples.

Reply:
Good catch. I hadn't thought about the metadata that the LSP spec allows us to display while cycling through autocomplete suggestions.

I have made the tree generic and documented it.